spec/index.html

```diff
@@ -283,9 +283,157 @@ <h2>Introduction</h2>
   <section id="basic-concepts" class="informative">
     <h2>Basic Concepts</h2>
 
-    <p>FIXME.</p>
+    <p>
+      To ease writing and collaborating on JSON-LD documents, it is a common practice
+      to serialize them as YAML.
+      This requires a registered media type, not only to enable content negotiation
+      of linked data documents in YAML, but even to define the expected behavior of
+      applications that processes these documents, including fragment identifiers and
+      interoperability considerations.
+    </p>
+
+    <p>
+      This is because YAML is more flexible than JSON:
+    </p>
+
+      <ul>
+        <li>YAML supports different encodings, including UTF-8, UTF-16, and UTF-32.</li>
+        <li>YAML supports more data types than JSON.</li>
+        <li>the structure of a YAML document &mdash; that is, a named YAML representation graph &mdash;
+        is a rooted, directed graph that can have cycles.</li>
+
+    <p>
+      The first goal of this specification is to allow a JSON-LD document to be
+      processed and serialized into YAML, and then back into JSON-LD, without
+      losing any information.
+
+      This is always possible, because a YAML representation graph can always represent
+      a tree, because JSON data types are a subset of YAML's, and because
+      JSON encoding is UTF-8.
+    </p>
+
+    <p data-format="markdown>Example: the JSON-LD document below
+
+    ```
+    {
+      "@context": "http://example.org/context.jsonld",
+      "@graph": [
+        {"@id": "http://example.org/1", "title": "Example 1"},
+        {"@id": "http://example.org/2", "title": "Example 2"},
+        {"@id": "http://example.org/3", "title": "Example 3"}
+      ]
+    }
+    ```
+
+    can be serialized as YAML as follows.
+    Note that entries 
+    starting with  `@` need to be enclosed in double quotes or escaped because
+    `@` is a reserved character in YAML.
+
+    ```yaml
+    %YAML 1.2
+    ---
+    "@context": http://example.org/context.jsonld
+    "@graph":
+      -
+        "@id": http://example.org/1
+        title: Example 1
+      -
+        "@id": http://example.org/2
+        title: Example 2
+      -
+        "@id": http://example.org/3
+        title: Example 3
+    ```
+
+
+    </p>
   </section>
+  <section id="specifications" class="normative">
+    <h2>Core Requirements</h2>
 
+    <p>
+      A YAML-LD document is a [[YAML]] document that can be interpreted as Linked Data [[LINKED-DATA]].
+    </p>
+    <p>
+      It MUST be encoded in UTF-8, to ensure interoperability with [[JSON]]. 
+    </p>
+    <p>
+      Comments in YAML-LD documents
+      MAY be removed when interpreting the document
```

Comment:
The MAY wording may be odd, since it implies that there is process that is optional. If so, what is the process for implementations wanting to preserve comments? And how could they possibly be round-tripped? It may be better to simply require that they be dropped.

Reply:
> ... the process is optional

I think that it is optional. One could e.g.

```
g = rdflib.Graph()
g.parse("foo.yamlld")
g.serialize("bar.yamlld")  #  retain comments
g.serialize("bar.ttl")         # retain comments ?
g.serialize("bar.jsonld")  # strip comment
```

> what is the process for ... preserving comments?

Maybe there's somebody out there that can do better than me, so I don't want to limit him.

> require that they be dropped

An implementation that is capable of preserving comments e.g. in its internal representation, or when converting YAML to Turtle would thus be non conformant.

Reply:
No other RDF serialization format that I'm aware of retains comments post-parsing. As with [Turtle](https://www.w3.org/TR/turtle/#sec-grammar-comments), we should probably just treat comments as white space. Normative behavior on ignoring them isn't necessary.

Are there _any_ examples of other serialization formats (RDF anyway) retaining comments? Isn't that the purpose of something like `rdfs:comment`, which specifically allows for a way of introducing non-syntactic comments in a document?

Reply:
ruamel.yaml is a YAML implementation that preserves comments, so I just don't want forbid something that's actually feasible, e.g.

```
# pip install ruamel.yaml
from ruamel.yaml import YAML
from sys import stdout

yaml = YAML()
data = yaml.load("""
# Ciao
a: 1
""")
yaml.dump(data, stdout)
```
I think that "treating as white space" is ok, without normative behavior.